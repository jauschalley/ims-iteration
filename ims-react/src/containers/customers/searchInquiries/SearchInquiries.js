import React from 'react';
import {observer} from "mobx-react";
import {Formik, Field} from 'formik';
import {Row, Container, Col, Form, FormLabel, Button} from 'react-bootstrap';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import CwigCard from '../../../components/CwigCard';

const mapSelectOptions = (options, labelKey, idKey) => (
  options.map((option) => (
    <option value={option[idKey]}>
      {option[labelKey]}
    </option>
  ))
);

const SearchInquiries = observer(({inquirySearchStore, referenceDataStore}) => {
  let refData = referenceDataStore.referenceData;

  return(
    <Container>
      <Row><h1 class="margin-30">Search Inquiries</h1></Row>
      <Formik
      initialValues={{
        inquiryNumber: '',
        firstName: '',
        lastName: '',
        email: '',
        state: '',
        assignedTo: '',
        dateTo: '',
        dateFrom: '',
        status: ''
      }}
      onSubmit={(values) => inquirySearchStore.searchInquiries(values)}
      >
        {formikProps =>
          <Form>
            <CwigCard>
              <Row>
                <Col md={4}>
                  <FormLabel>Inquiry Number</FormLabel>
                  <Field placeholder="Inquiry Number" name="inquiryNumber"></Field>
                </Col>
                <Col md={4}>
                <FormLabel>First Name</FormLabel>
                  <Field placeholder="First Name" name="firstName"></Field>
                </Col>
                <Col md={4}>
                  <FormLabel>Last Name</FormLabel>
                  <Field placeholder="Last Name" name="lastName"></Field>
                </Col>
              </Row>
              <Row>
                <Col md={4}>
                  <FormLabel>Email</FormLabel>
                  <Field placeholder="Email" name="email"></Field>
                </Col>
                <Col md={4}>
                  <FormLabel>State</FormLabel>
                  <Field as="select" name="state">
                    {mapSelectOptions(refData.states, 'state', 'stateID')}
                  </Field>
                </Col>
                <Col md={4}>
                  <FormLabel>Assigned To</FormLabel>
                  <Field placeholder="Assigned To" name="assignedTo"></Field>
                </Col>
              </Row>
              <Row>
                <Col md={6}>
                  <FormLabel>Date Range</FormLabel>
                  <Field placeholder="Date From" name="dateFrom"></Field>
                  <Field placeholder="Date To" name="dateTo"></Field>
                </Col>
                <Col md={4}>
                  <FormLabel>Status</FormLabel>
                  <Field as="select" name="status">
                    <option value='0'>Open</option>
                    <option value='1'>Closed</option>
                  </Field>
                </Col>
              </Row>
              <Row>
                <Button className="search" onClick={formikProps.handleSubmit}>Search</Button>
              </Row>
            </CwigCard>
          </Form>
        }
      </Formik>
      {inquirySearchStore.searchResults && <TableContainer component={Paper}>
        <Table aria-label="search customers table">
          <TableHead>
            <TableRow>
              <TableCell>Inquiry ID</TableCell>
              <TableCell>Name</TableCell>
              <TableCell>Organization</TableCell>
              <TableCell>Email</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {inquirySearchStore.searchResults.map(row => (
              <TableRow key={row.name}>
                <TableCell>{row.inquiryID}</TableCell>
                <TableCell>{row.firstName} {row.lastNameName}</TableCell>
                <TableCell>{row.organization}</TableCell>
                <TableCell>{row.email}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>}
    </Container>
  );
});

export default SearchInquiries;