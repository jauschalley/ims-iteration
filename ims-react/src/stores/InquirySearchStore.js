import {observable, action, decorate} from "mobx";
import RequestService from '../services/RequestService';
import config from '../config';

class InquirySearchStore {
  searchResults = null;

  searchInquiries = (formData) => {
    RequestService.post(`${config.backendUrl}/inquiries`, formData, (response) => {
      this.searchResults = response.data.inquiries;
    });
  }
}

decorate(InquirySearchStore, {
  searchResults: observable,
  searchInquiries: action
});

export default InquirySearchStore;