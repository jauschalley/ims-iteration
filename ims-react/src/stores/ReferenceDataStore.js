import {observable, decorate} from "mobx";

class ReferenceDataStore {
  referenceData = null;

  constructor(referenceData = null) {
    this.referenceData = referenceData || require('../data/lookups.json');
  }
}

decorate(ReferenceDataStore, {
  referenceData: observable,
});

export default ReferenceDataStore;