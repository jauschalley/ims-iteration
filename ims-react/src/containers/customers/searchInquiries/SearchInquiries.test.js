import React from 'react';
import renderer from 'react-test-renderer';
import ReferenceDataStore from '../../../stores/ReferenceDataStore';
import SearchInquiries from './SearchInquiries';
import referenceData from '../newInquiry/mockReferenceData';
import InquirySearchStore from '../../../stores/InquirySearchStore';
import mockSearchResults from './mockSearchResults';

describe('<SearchInquiries />', () => {
  it('renders correctly', () => {
    const referenceDataStore = new ReferenceDataStore(referenceData);
    const tree = renderer
      .create(
        <SearchInquiries 
          referenceDataStore={referenceDataStore}
          inquirySearchStore={new InquirySearchStore(mockSearchResults)}
        />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
