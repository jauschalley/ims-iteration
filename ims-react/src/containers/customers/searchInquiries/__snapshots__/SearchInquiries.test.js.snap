// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInquiries /> renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <h1>
      Search Inquiries
    </h1>
  </div>
  <form
    className=""
  >
    <div
      className="cwig-card container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Inquiry Number
          </label>
          <input
            name="inquiryNumber"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Inquiry Number"
            value=""
          />
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            First Name
          </label>
          <input
            name="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="First Name"
            value=""
          />
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Last Name
          </label>
          <input
            name="lastName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Last Name"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Email
          </label>
          <input
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email"
            value=""
          />
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            State
          </label>
          <select
            name="state"
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          >
            <option
              value={1}
            >
              Alabama
            </option>
            <option
              value={2}
            >
              Alaska
            </option>
            <option
              value={3}
            >
              American Samoa
            </option>
            <option
              value={4}
            >
              Arizona
            </option>
            <option
              value={5}
            >
              Arkansas
            </option>
            <option
              value={6}
            >
              California
            </option>
            <option
              value={7}
            >
              Colorado
            </option>
            <option
              value={8}
            >
              Connecticut
            </option>
            <option
              value={9}
            >
              Delaware
            </option>
            <option
              value={10}
            >
              District of Columbia
            </option>
            <option
              value={11}
            >
              Florida
            </option>
            <option
              value={12}
            >
              Georgia
            </option>
            <option
              value={13}
            >
              Guam
            </option>
            <option
              value={14}
            >
              Hawaii
            </option>
            <option
              value={15}
            >
              Idaho
            </option>
            <option
              value={16}
            >
              Illinois
            </option>
            <option
              value={17}
            >
              Indiana
            </option>
            <option
              value={18}
            >
              Iowa
            </option>
            <option
              value={19}
            >
              Kansas
            </option>
            <option
              value={20}
            >
              Kentucky
            </option>
            <option
              value={21}
            >
              Louisiana
            </option>
            <option
              value={22}
            >
              Maine
            </option>
            <option
              value={23}
            >
              Marshall Islands
            </option>
            <option
              value={24}
            >
              Maryland
            </option>
            <option
              value={25}
            >
              Massachusetts
            </option>
            <option
              value={26}
            >
              Michigan
            </option>
            <option
              value={27}
            >
              Minnesota
            </option>
            <option
              value={28}
            >
              Mississippi
            </option>
            <option
              value={29}
            >
              Missouri
            </option>
            <option
              value={30}
            >
              Montana
            </option>
            <option
              value={31}
            >
              Nebraska
            </option>
            <option
              value={32}
            >
              Nevada
            </option>
            <option
              value={33}
            >
              New Hampshire
            </option>
            <option
              value={34}
            >
              New Jersey
            </option>
            <option
              value={35}
            >
              New Mexico
            </option>
            <option
              value={36}
            >
              New York
            </option>
            <option
              value={37}
            >
              North Carolina
            </option>
            <option
              value={38}
            >
              North Dakota
            </option>
            <option
              value={39}
            >
              Northern Mariana Islands
            </option>
            <option
              value={40}
            >
              Ohio
            </option>
            <option
              value={41}
            >
              Oklahoma
            </option>
            <option
              value={42}
            >
              Oregon
            </option>
            <option
              value={43}
            >
              Palau
            </option>
            <option
              value={44}
            >
              Pennsylvania
            </option>
            <option
              value={45}
            >
              Puerto Rico
            </option>
            <option
              value={46}
            >
              Rhode Island
            </option>
            <option
              value={47}
            >
              South Carolina
            </option>
            <option
              value={48}
            >
              South Dakota
            </option>
            <option
              value={49}
            >
              Tennessee
            </option>
            <option
              value={50}
            >
              Texas
            </option>
            <option
              value={51}
            >
              Utah
            </option>
            <option
              value={52}
            >
              Vermont
            </option>
            <option
              value={53}
            >
              Virgin Islands
            </option>
            <option
              value={54}
            >
              Virginia
            </option>
            <option
              value={55}
            >
              Washington
            </option>
            <option
              value={56}
            >
              West Virginia
            </option>
            <option
              value={57}
            >
              Wisconsin
            </option>
            <option
              value={58}
            >
              Wyoming
            </option>
          </select>
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Assigned To
          </label>
          <input
            name="assignedTo"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Assigned To"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <label
            className="form-label"
          >
            Date Range
          </label>
          <input
            name="dateFrom"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Date From"
            value=""
          />
          <input
            name="dateTo"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Date To"
            value=""
          />
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Status
          </label>
          <select
            name="status"
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          >
            <option
              value="0"
            >
              Open
            </option>
            <option
              value="1"
            >
              Closed
            </option>
          </select>
        </div>
      </div>
      <div
        className="row"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        >
          Search
        </button>
      </div>
    </div>
  </form>
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="search customers table"
      className="MuiTable-root"
    >
      <thead
        className="MuiTableHead-root"
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Inquiry ID
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Organization
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Email
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
      >
        <tr
          className="MuiTableRow-root"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            56944
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            
             
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            56912
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            
             
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            56334
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            
             
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            55305
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            Pawfj
             
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            hfjfeyatj@dsmmtlsaqnms.cvx
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            55304
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            Qkvj
             
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            ihkfxuoxto@uamvj.exr
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            55289
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            Tftnvfq
             
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            iquhgv.fppbkcco@hiq.sum
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            52528
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            Mltjvadlw
             
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
